src/model/page-config-resolver-test.js

```diff
@@ -268,7 +272,337 @@ describes.realWin('PageConfigResolver', {}, env => {
+      });
+
+      const validAccessElement = createElement(doc, 'div');
+      validAccessElement.innerHTML =
+          '<div itemprop="hasPart" itemscope itemtype="http://schema.org/WebPageElement"> \
+            <meta itemprop="isAccessibleForFree" content=false/> \
+            <meta itemprop="cssSelector" content=".paywalled-section"/> \
+          </div>';
+      doc.getElementById('top').appendChild(validAccessElement);
+
+      return resolver.resolveConfig().then(config => {
+        expect(config.isLocked()).to.be.true;
+        expect(config.getProductId()).to.equal('pub1:premium');
+      });
+    });
+
+    it('should wait until dom is ready', () => {
+      const resolver = new PageConfigResolver(gd);
+      // Add content.
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+          '<div itemscope itemtype="http://schema.org/NewsArticle"> \
+            <meta itemprop="isAccessibleForFree" content="False"/> \
+            <div itemprop="isPartOf" itemscope itemtype="http://schema.org/CreativeWork http://schema.org/Product"> \
+              <meta itemprop="name" content="New York Times"/> \
+              <meta itemprop="productID" content="pub1:premium"/> \
+            </div> \
+            <div itemprop="hasPart" itemscope itemtype="http://schema.org/WebPageElement"> \
+              <meta itemprop="isAccessibleForFree" content=true/> \
+              <meta itemprop="cssSelector" content=".paywalled-section"/> \
+            </div> \
+            <div itemprop="articleBody" class="paywalled-section"> \
+              Paid content possibly. \
+            </div> \
+          </div>';
+      addMicrodata(divElement);
+
+      // Document not complete and no siblings.
+      let config = resolver.check();
+      expect(config).to.be.null;
+
+      // Complete document.
+      readyState = 'complete';
+      config = resolver.check();
+      expect(config).to.be.ok;
+      expect(config.getProductId()).to.equal('pub1:premium');
+      return expect(resolver.resolveConfig()).to.eventually.equal(config);
+    });
+
+    it('malformed microdata no productId', () => {
+      // Add content.
+      const divElement = createElement(doc, 'div');
+      divElement.innerHTML =
+          '<div itemscope itemtype="http://schema.org/NewsArticle"> \
+            <div itemprop="hasPart" itemscope itemtype="http://schema.org/WebPageElement"> \
+              <meta itemprop="isAccessibleForFree" content=true/> \
+              <meta itemprop="cssSelector" content=".paywalled-section"/> \
+            </div> \
+            <div itemprop="articleBody" class="paywalled-section"> \
+              Paid content possibly. \
+            </div> \
+          </div>';
+      addMicrodata(divElement);
+      readyState = 'complete';
+      const resolver = new PageConfigResolver(gd);
+      expect(resolver.check()).to.be.null;
+    });
+
+    it('malformed microdata no access info default to unlocked', () => {
```

Comment:
That's not "malformed" - an expect case. Though I thought you tested it plenty above.

Reply:
will remove this case.